client.go

```diff
@@ -157,69 +158,25 @@ func (c *TunaSessionClient) newTunaExit(i int) (*tuna.TunaExit, error) {
 }
 
 func (c *TunaSessionClient) Listen(addrsRe *nkngomobile.StringArray) error {
-	acceptAddrs, err := getAcceptAddrs(addrsRe)
-	if err != nil {
-		return err
-	}
-
 	c.Lock()
 	defer c.Unlock()
 
-	c.acceptAddrs = acceptAddrs
-
 	if len(c.listeners) > 0 {
 		return nil
 	}
 
-	if c.wallet == nil {
-		return errors.New("wallet is empty")
-	}
-
-	listeners := make([]net.Listener, c.config.NumTunaListeners)
-	for i := 0; i < len(listeners); i++ {
-		listeners[i], err = net.Listen("tcp", "127.0.0.1:")
-		if err != nil {
-			return err
-		}
+	acceptAddrs, err := getAcceptAddrs(addrsRe)
```

Comment:
This should be before checking listeners. Multiple call to Listen() should update accept addrs